functorch/experimental/output_dtype.py

```diff
@@ -0,0 +1,136 @@
+from functools import partial 
+
+import torch
+import torch.utils._pytree as pytree
+from torch.fx.experimental.proxy_tensor import (
+    ProxyTorchDispatchMode,
+    disable_proxy_modes_tracing,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+    unwrap_proxy,
+)
+from torch.utils._python_dispatch import (
+    _get_current_dispatch_mode,
+    _pop_mode_temporarily,
+)
+from torch._C import DispatchKey
+from torch._functorch.eager_transforms import (
+    _unwrap_all_tensors_from_functional,
+    _wrap_all_tensors_to_functional,
+)
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+
+
+output_dtype = HigherOrderOperator("output_dtype")
+output_dtype.fallthrough(DispatchKey.PythonDispatcher)
+output_dtype.fallthrough(DispatchKey.PythonTLSSnapshot)
+output_dtype.fallthrough(DispatchKey.ADInplaceOrView)
+output_dtype.fallthrough(DispatchKey.BackendSelect)
+output_dtype.fallthrough(DispatchKey.AutocastCPU)
```

Comment:
Why are some of these fallthrough while others have impls defined?

Reply:
Bc we don't have any specific things we have to implement for those keys.